src/keboola/component/table_schema.py

```diff
@@ -0,0 +1,93 @@
+from typing import List, Dict
+from typing import Optional, Union
+from keboola.component.dao import SupportedDataTypes
+import os
+import json
+from dataclasses import dataclass
+
+
+@dataclass
+class FieldSchema:
+    """
+    Defines the name and type specifications of a single field in a table
+    """
+    name: str
+    base_type: Optional[Union[SupportedDataTypes, str]] = None
+    description: Optional[str] = None
+    nullable: bool = False
+    length: Optional[str] = None
+    default: Optional[str] = None
+
+
+@dataclass
+class TableSchema:
+    """
+    TableSchema class is used to define the schema and metadata of a table.
+    """
+    name: str
+    fields: List[FieldSchema]
+    primary_keys: Optional[List[str]] = None
+    parent_tables: Optional[List[str]] = None
+    description: Optional[str] = None
+
+    @property
+    def field_names(self) -> List[str]:
+        return [column.name for column in self.fields]
+
+    @property
+    def csv_name(self) -> str:
+        return f"{self.name}.csv"
+
+
+def _init_table_schema_from_dict(json_table_schema: Dict) -> TableSchema:
```

Comment:
I can imagine this being a public method

Reply:
Agreed, added in next commit